src/Reduce.php

```diff
@@ -154,4 +154,36 @@ public static function toString(iterable $data, string $separator = '', string $
         $joined = \implode($separator, $items);
         return $prefix . $joined . $suffix;
     }
+
+    /**
+     * Reduces given collection to array of its upper and lower bounds.
+     *
+     * Returns [null, null] if given collection is empty.
+     *
+     * @param iterable<numeric> $numbers
+     *
+     * @return array{numeric, numeric}|array{null, null}
+     */
+    public static function toBounds(iterable $numbers): array
+    {
+        return static::toValue($numbers, static function ($carry, $datum) {
+            return [\min($carry[0] ?? $datum, $datum), \max($carry[1] ?? $datum, $datum)];
+        }, [null, null]);
+    }
+
+    /**
+     * Reduces given collection to its amplitude.
+     *
+     * Returns 0 if given collection is empty.
+     *
+     * @param iterable<numeric> $numbers
+     *
+     * @return int|float
+     */
+    public static function toAmplitude(iterable $numbers)
```

Comment:
Suggestion: Let's call this `toRange`. Range is the max - min. Amplitude is usually half the distance of the range. For a wave, it is from x = 0 to the peek.

See https://www.mathwords.com/a/amplitude.htm

Reply:
Done.